### PR TITLE
Feature/config file

### PR DESCRIPTION
**Pull request by CatarinaCarolina**
make a config file with constants that can be adjusted by the user

adjust LCS and Extend checks, separate methodology between region and protos, so that region follows the behaviour of V1, and protos need a biosynthetic domain in the LCS, and a minimum of 3 domains. exceptions include antismash classes for which only 1 domain is required to render a BGC in said class, such as terpenes.

calculate a comparable region via LCS+EXT only in modes GLOCAL or AUTO with contig edge.

### CODE REVIEW

File: big_scape/cli/config.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        # write config log
+        BigscapeConfig.write_config_log(run, config)
+
+    @staticmethod
+    def write_config_log(run, config) -> None:
+        """_summary_
+
+        Args:
+            log_path (Path): path to log file
```
Comment (nlouwen) [inline big_scape/cli/config.py:80]: small docstring detail, _summary_ and args

File: big_scape/comparison/workflow.py [hunk 2]
```diff
@@ -227,49 +226,83 @@ def do_lcs_pair(
...
-    if check(pair, 3, True):
-        return True
-
-    logging.debug("resetting after extend")
+    else:
+        # Proto LCS: must contain biosynthetic domain, no min length
+        if len_check(pair, BigscapeConfig.PROTO_MIN_LCS_LEN) or biosynthetic_check(
+            pair
+        ):
+            return True
```
Comment (nlouwen) [inline big_scape/comparison/workflow.py:249]: change to 'and' if it must contain biosynth

File: big_scape/comparison/extend.py [hunk 3]
```diff
@@ -44,25 +44,26 @@ def check(pair: RecordPair, min_len: int, biosynth_check: bool) -> bool:
...
+        return True
+
+    a_len = pair.comparable_region.a_stop - pair.comparable_region.a_start
+    b_len = pair.comparable_region.b_stop - pair.comparable_region.b_start
+
+    if a_len >= min_len and b_len >= min_len:
+        return True
+
+    return False
+
```
Comment (nlouwen) [inline big_scape/comparison/extend.py:57]: recheck if this check function can be completely removed, does not seem to be used anymore

Comment (CatarinaCarolina) [reply big_scape/comparison/extend.py:57]: removed it

**Review by nlouwen — APPROVED** [review]

**Pull request merged**